tests/protos_append/builder_ext.rs

```diff
@@ -360,3 +361,71 @@ impl VectorsSelector {
         }
     }
 }
+
+impl RecommendPointsBuilder {
+    /// Look for vectors closest to the vectors from these points or vectors
+    pub fn positive(&mut self, recommend_example: impl Into<RecommendExample>) -> &mut Self {
+        let recommend_example = recommend_example.into();
+        match recommend_example {
+            RecommendExample::PointId(point_id) => {
+                let mut vec = self.positive.take().unwrap_or_default();
+                vec.push(point_id);
+                self.positive = Some(vec);
```

Comment:
What about something like this:

```rust
if let Some(vec) = self.positive.get_or_insert_with(Vec::new) {
    vec.push(point_id);
}
```

I haven't tested it, but expect something like that to work. That prevents taking out and reinserting the vector. Of course, the same applies to the other changes in this file.

There's also [`get_or_insert_default()`](https://doc.rust-lang.org/core/option/enum.Option.html#method.get_or_insert_default) but it seems to be Nightly only at this time.

Reply:
That is indeed cleaner. Thanks for pointing this out!
Fixed now